lib/spack/spack/cmd/bootstrap.py

```diff
@@ -15,13 +17,48 @@
 import spack.cmd.common.arguments
 import spack.config
 import spack.main
+import spack.mirror
+import spack.spec
+import spack.stage
 import spack.util.path
 
 description = "manage bootstrap configuration"
 section = "system"
 level = "long"
 
 
+# Tarball to be downloaded if binary packages are requested in a local mirror
+BINARY_TARBALL = 'https://github.com/alalazo/spack-bootstrap-mirrors/releases/download/v0.1-rc.2/bootstrap-buildcache.tar.gz'
+
+#: Subdirectory where to create the mirror
+LOCAL_MIRROR_DIR = 'local-mirror'
```

Comment:
This should be an option to the `bootstrap mirror` command, and the binary metadata can be templated by the chosen directory

Reply:
We should probably be able to specify a mirror to add the bootstrapping binaries into -- in the long run, we'll starat bootstrapping things besides `gpg` and `clingo`, and they will behave more like normal binaries. We should make it easy to reuse those in the mirror if they're already present.

Reply:
> This should be an option to the bootstrap mirror command, and the binary metadata can be templated by the chosen directory

This will add complexity to give people the opportunity to change a name. I'd prefer to just use a different name if we have a better one.

Reply:
> We should probably be able to specify a mirror to add the bootstrapping binaries into -- in the long run, we'll starat bootstrapping things besides gpg and clingo, and they will behave more like normal binaries. We should make it easy to reuse those in the mirror if they're already present.

Besides GnuPG, clingo and patchelf, I think all the other bootstrapping we need should go through signed binary packages, as we do for all other software. I don't know which direction we'll take in the future to reduce the complexity imposed by these 3 packages that are verified by their sha256 sum, so I'm hesitant to provide a lot of options here. I'd prefer to keep this hard-coded and simple and eventually extend the command as soon as we give people simple ways to create a bootstrapping mirror on their own. Would that be acceptable?

Reply:
I think I was confused earlier, as I read below I see there is an option for the directory to store the whole thing in, and this is an internal dirname inside that.

I think this name should probably be `bootstrap_cache` (to match `binary_cache` for normal binaries).